File: 1000/139A/main.cpp
#include <iostream>
using namespace std;

int main() {
    int pages;
    cin >> pages;

    int days[7];
    int sum = 0;
    for(int i = 0; i < 7; i++) 
    {
        cin >> days[i];
        sum += days[i];
    }

    pages = pages % sum;
    if(pages == 0)
        pages = sum;

    int ans = 0;
    int i = 0;
    while(true) 
    {
        if(pages <= days[i]) 
        {
            ans = i + 1;
            break;
        }
        
        pages -= days[i];
        i = (i + 1) % 7;
    }

    cout << ans << endl;
    return 0;
}

File: 1000/139A/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int pages = sc.nextInt();
        
        int days[] = new int[7];
        int sum = 0;
        for(int i=0;i<7;i++)
        {
            days[i] = sc.nextInt();
            sum += days[i];
        }
        
        pages = pages % sum;
        if(pages == 0)
            pages = sum;
        
        int ans = 0;
        int i = 0;
        
        for(;i<7;)
        {
            if(pages <= days[i])
            {
                ans = i+1;
                break;
            }

            pages -= days[i];
            i = (i+1)%7;
        }
        
        System.out.println(ans);
    }
}